strax/context.py

```diff
@@ -1656,6 +1655,52 @@ def copy_to_frontend(self,
                     f'Trying to write {data_key} to {t_sf} which already exists, '
                     'do you have two storage frontends writing to the same place?')
 
+    def get_source(self,
+                   run_id: str,
+                   target: str,
+                   check_forbidden: bool = True,
+                   ) -> ty.Union[set, None]:
+        """
+        For a given run_id and target get the stored bases where we can
+        start processing from, if no base is available, return None.
+
+        :param run_id: run_id
+        :param target:  target
+        :param check_forbidden: Check that we are not requesting to make
+            a plugin that is forbidden by the context to be created.
+        :return: set of plugin names that are needed to start processing
+            from and are needed in order to build this target.
+        """
+        if self.is_stored(run_id, target):
+            return {target}
+
+        deps = strax.to_str_tuple(self._plugin_class_registry[target].depends_on)
+        if not deps:
+            return None
+
+        forbidden = strax.to_str_tuple(self.context_config['forbid_creation_of'])
+        if check_forbidden and target in forbidden:
+            return None
```

Comment:
Maybe be more explicit in this case and raise an error or make a warning. Like you have further down.

Reply:
good point!